src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1039,24 +1108,58 @@ namespace cryptonote
       if(tvc[i].m_added_to_pool)
         MDEBUG("tx added: " << results[i].hash);
     }
+
+    // relay those that are valid
+    NOTIFY_NEW_TRANSACTIONS::request r;
+    it = m_defered_txs.begin();
+    CHECK_AND_ASSERT_MES(m_defered_txs.size() == tvc.size(), false, "m_defered_txs and tvc do not have the same size");
+    for (size_t i = 0; i < m_defered_txs.size(); i++, ++it) {
+      m_defered_txs[i].on_verified(!tvc[i].m_verifivation_failed);
+      if (!tvc[i].m_verifivation_failed && tvc[i].m_should_be_relayed && !already_have[i])
+        r.txs.push_back(it->blob.blob);
+    }
+    if (!r.txs.empty())
+    {
+      cryptonote_connection_context fake_context = AUTO_VAL_INIT(fake_context);
+      get_protocol()->relay_transactions(r, fake_context);
```

Comment:
Will we relay a tx to its origin connection too?

Reply:
Very good point. This is pointless, I'll change.

Reply:
Except I purposefully did not add the context there since core has no business knowing it, so it's a callback. Since this is harmless (beyond a bit of extra bandwidth), I'll leave it that way for now and include in the bandwidth saving patches.